streamconfig/config.go

```diff
@@ -43,23 +61,78 @@ type Producer struct {
 
 	// Logger is the configurable logger instance to log messages. If left
 	// undefined, a no-op logger will be used.
-	Logger zap.Logger
+	Logger zap.Logger `ignored:"true"`
 
 	// HandleInterrupt determines whether the producer should close itself
 	// gracefully when an interrupt signal (^C) is received. This defaults to true
 	// to increase first-time ease-of-use, but if the application wants to handle
 	// these signals manually, this flag disables the automated implementation.
-	HandleInterrupt bool
+	HandleInterrupt bool `ignored:"true"`
+
+	// Name is the name of the current processor. It is currently only used to
+	// determine the prefix for environment-variable based configuration values.
+	// For example, if Name is set to `MyProcessor`, then all environment
+	// variable-based configurations need to start with `MYPROCESSOR_`.If no name
+	// is set, then the prefix is "producer" is used, so you prepend all
+	// environment variables with "PRODUCER_.
+	Name string `ignored:"true"`
+
+	// AllowEnvironmentBasedConfiguration allows you to disable configuring the
+	// stream client based on predefined environment variables. This is enabled by
+	// default, but can be disabled if you want full control over the behavior of
+	// the stream client without any outside influence.
+	AllowEnvironmentBasedConfiguration bool `ignored:"true"`
 }
 
 // ConsumerDefaults holds the default values for Consumer.
 var ConsumerDefaults = Consumer{
 	Logger:          *zap.NewNop(),
 	HandleInterrupt: true,
+	Name:            "consumer",
+	AllowEnvironmentBasedConfiguration: true,
 }
 
 // ProducerDefaults holds the default values for Producer.
 var ProducerDefaults = Producer{
 	Logger:          *zap.NewNop(),
 	HandleInterrupt: true,
+	Name:            "producer",
+	AllowEnvironmentBasedConfiguration: true,
+}
+
+// Usage calls the usage() function and returns a byte array.
```

Comment:
This is a pretty vague description, in order to understand it I need to read what `usage()` does. How about:

> Usage returns an explanation on how to set the configuration values using environment variables.

Reply:
Haha, yeah, I had the same description twice on both methods before, but moved it to the `usage` function to do some DRY'ing, but that was a bad decision, since that won't show up in editors for example. Will fix.

Reply:
https://github.com/blendle/go-streamprocessor/pull/60/commits/345a24e01e3cf97a700baa95b50acd8fcbcc9d11